File: ui-ngx/src/app/core/http/usage-info.service.ts
// SPDX-FileCopyrightText: Copyright The Thingsboard Authors
// SPDX-License-Identifier: Apache-2.0
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { defaultHttpOptionsFromConfig, RequestConfig } from '@core/http/http-utils';
import { UsageInfo } from '@shared/models/usage.models';

@Injectable({
  providedIn: 'root'
})
export class UsageInfoService {

  constructor(
    private http: HttpClient
  ) {}

  public getUsageInfo(config?: RequestConfig): Observable<UsageInfo> {
    return this.http.get<UsageInfo>('/api/usage',
      defaultHttpOptionsFromConfig(config));
  }

}

File: common/dao-api/src/main/java/org/thingsboard/server/dao/usage/UsageInfoService.java
// SPDX-FileCopyrightText: Copyright The Thingsboard Authors
// SPDX-License-Identifier: Apache-2.0
package org.thingsboard.server.dao.usage;

import org.thingsboard.server.common.data.UsageInfo;
import org.thingsboard.server.common.data.id.TenantId;

public interface UsageInfoService {

    UsageInfo getUsageInfo(TenantId tenantId);

}
